### PR TITLE
Add ingress status integration test

### PR DESCRIPTION
**Pull request by Monkeyanator**
Adds an integration test to make sure ingress status is updated for ingresses we're handling (#28225) 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/pilot/ingress_test.go [hunk 1]
```diff
@@ -257,5 +260,23 @@ spec:
 					apps.Ingress.CallEchoWithRetryOrFail(ctx, c.call, retry.Timeout(time.Minute*2))
 				})
 			}
+
+			t.Run("status", func(t *testing.T) {
+				if !ctx.Environment().(*kube.Environment).Settings().LoadBalancerSupported {
```
Comment (Monkeyanator) [inline tests/integration/pilot/ingress_test.go:265]: noticed that prow runs pilot tests w/ the `--istio.test.loadbalancer=false`, so where does this test get run?

Comment (howardjohn) [reply tests/integration/pilot/ingress_test.go:265]: Ah I did not know that, I thought we were deploying metallb (allows LB on kind) for everything but looks like just for multicluster. We do have MC tests that run these which would trigger it I believe, although I think they are still WIP and hidden currently

**Review by howardjohn — APPROVED** [review]

**Pull request merged**